### PR TITLE
use unordered containers where possible (patch 4 on X)

### PR DESCRIPTION
**Pull request by nerzhul**
Also remove some unused parameters/functions

### CODE REVIEW

File: src/content_sao.cpp [hunk 1]
```diff
@@ -347,8 +347,10 @@ void LuaEntitySAO::step(float dtime, bool send_recommended)
 
 	if(m_bone_position_sent == false){
 		m_bone_position_sent = true;
-		for(std::map<std::string, core::vector2d<v3f> >::const_iterator ii = m_bone_position.begin(); ii != m_bone_position.end(); ++ii){
-			std::string str = gob_cmd_update_bone_position((*ii).first, (*ii).second.X, (*ii).second.Y);
+		for (UNORDERED_MAP<std::string, core::vector2d<v3f> >::const_iterator
+				 ii = m_bone_position.begin(); ii != m_bone_position.end(); ++ii){
```
Comment (est31) [inline src/content_sao.cpp]: mixed indentation. Please only use tabs

File: src/content_sao.cpp [hunk 2]
```diff
@@ -383,8 +385,10 @@ std::string LuaEntitySAO::getClientInitializationData(u16 protocol_version)
 		os<<serializeLongString(gob_cmd_update_armor_groups(m_armor_groups)); // 2
 		os<<serializeLongString(gob_cmd_update_animation(
 			m_animation_range, m_animation_speed, m_animation_blend, m_animation_loop)); // 3
-		for(std::map<std::string, core::vector2d<v3f> >::const_iterator ii = m_bone_position.begin(); ii != m_bone_position.end(); ++ii){
-			os<<serializeLongString(gob_cmd_update_bone_position((*ii).first, (*ii).second.X, (*ii).second.Y)); // m_bone_position.size
+		for (UNORDERED_MAP<std::string, core::vector2d<v3f> >::const_iterator
+				 ii = m_bone_position.begin(); ii != m_bone_position.end(); ++ii) {
```
Comment (est31) [inline src/content_sao.cpp]: same here

File: src/content_sao.cpp [hunk 3]
```diff
@@ -1007,19 +1012,22 @@ void PlayerSAO::step(float dtime, bool send_recommended)
 		m_messages_out.push(aom);
 	}
 
-	if(m_bone_position_sent == false){
+	if (!m_bone_position_sent) {
 		m_bone_position_sent = true;
-		for(std::map<std::string, core::vector2d<v3f> >::const_iterator ii = m_bone_position.begin(); ii != m_bone_position.end(); ++ii){
-			std::string str = gob_cmd_update_bone_position((*ii).first, (*ii).second.X, (*ii).second.Y);
+		for (UNORDERED_MAP<std::string, core::vector2d<v3f> >::const_iterator
+				 ii = m_bone_position.begin(); ii != m_bone_position.end(); ++ii) {
```
Comment (est31) [inline src/content_sao.cpp]: and here

Comment (nerzhul) [inline src/content_sao.cpp]: right this is clion

**Comment by sfan5** [comment]
:+1: except est's comments

**Comment by nerzhul** [comment]
Changes fixed @est31

**Comment by sfan5** [comment]
155288ee981c70f505526347cb2bcda4df1c8e6b

**Pull request closed without merging**